src/Lykke.Service.MarketMakerReports.AzureRepositories/InventorySnapshotRepository.cs

```diff
@@ -13,12 +13,55 @@ namespace Lykke.Service.MarketMakerReports.AzureRepositories
     public class InventorySnapshotRepository : IInventorySnapshotRepository
     {
         private readonly INoSQLTableStorage<InventorySnapshotEntity> _storage;
+        
+        /// <summary>
+        /// IndexStorage is used to quickly get the last Snapshot for any date.
+        /// 
+        /// IndexStorage keeps references for every day to the last InventorySnapshot
+        /// for that particular day.
+        /// 
+        /// E.g. if the last snapshot in 2018-08-01 was received at 20:00:00, then
+        /// index entry for that date must be as follows:
+        /// 
+        /// PartitionKey | RowKey | ReferencePartitionKey | ReferenceRowKey
+        ///  2018-08-01  |  last  |   2018-08-01T20:00    | {id of the entry}
+        /// </summary>
+        private readonly INoSQLTableStorage<InventorySnapshotIndexEntity> _indexStorage;
 
-        public InventorySnapshotRepository(INoSQLTableStorage<InventorySnapshotEntity> storage)
+        public InventorySnapshotRepository(INoSQLTableStorage<InventorySnapshotEntity> storage,
+            INoSQLTableStorage<InventorySnapshotIndexEntity> indexStorage)
         {
             _storage = storage;
+            _indexStorage = indexStorage;
+        }
+
+        public async Task<IEnumerable<InventorySnapshot>> GetAsync(DateTime startDate, DateTime endDate)
+        {
+            var entities = await GetEntitiesAsync(startDate, endDate);
+
+            return entities.Select(x => JsonConvert.DeserializeObject<InventorySnapshot>(x.Json));
+        }
+
+        private async Task<IEnumerable<InventorySnapshotEntity>> GetEntitiesAsync(DateTime startDate, DateTime endDate)
```

Comment:
Do not mix public and private methods, put private method behind public methods

Reply:
Public methods are placed on top of the class.